libnavui-shield/src/main/java/com/mapbox/navigation/ui/shield/model/RouteShieldResult.kt

```diff
@@ -0,0 +1,13 @@
+package com.mapbox.navigation.ui.shield.model
+
+/**
+ * Data structure that wraps the information contained in [RouteShield] and [RouteShieldOrigin]
+ * and is used to render the shield.
+ *
+ * @property shield [RouteShield]
+ * @property origin [RouteShieldOrigin]
+ */
+data class RouteShieldResult(
```

Comment:
The same question regarding `data class` here.

Reply:
Resolved.

Reply:
By the way, can we do something like this: create a `public sealed interface` and a `internal data class` which implements it? Thus users will have access only to the properties and methods declared in the interface. And since it's sealed, they won't be able to inherit the interface and we can safely add new properties and methods to it. And we won't have to manually implement equals, hashCode and toString.